presto-main/src/main/java/com/facebook/presto/sql/gen/AndCodeGenerator.java

```diff
@@ -36,68 +38,71 @@ public BytecodeNode generateExpression(BytecodeGeneratorContext generator, Type
     {
         Preconditions.checkArgument(arguments.size() == 2);
 
-        Variable wasNull = generator.wasNull();
+        // We flatten the AND here.
+        Stack<RowExpression> stack = new Stack<RowExpression>();
+        stack.push(arguments.get(1));
+        stack.push(arguments.get(0));
+
+        List<RowExpression> flattenedArgs = new ArrayList<RowExpression>();
```

Comment:
`flattenedArguments`

Reply:
Done and also using the ImmutableList. Forgot about that earlier